how-to/postgrest.md

```diff
@@ -0,0 +1,262 @@
+postgrest memo.conf
+```
+
+Oh, for the installation method of PostgREST, please refer to https://postgrest.org/en/stable/explanations/install.html 😉
+
+## Access the URL
+Open your browser and access the following:
+
+http://localhost:3000/memos
+
+Result:
+
+```json
+[{"id":1,"title":"PostgreSQLはリレーショナル・データベース管理システムです。","content":"すごいでしょう"}, 
+ {"id":2,"title":"Groongaは日本語対応の高速な全文検索エンジンです。","content":"スワイショウ"}, 
+ {"id":3,"title":"PGroongaはインデックスとしてGroongaを使うためのPostgreSQLの拡張機能です。","content":"ハバナイスデー"}, 
+ {"id":4,"title":"groongaコマンドがあります。","content":"今日はコンバンワこのくにわ"}]
+```
+
+What? Is it really this easy to have a working REST API endpoint? It's too convenient! 🤯
+
+(However, please note that due to the permissions set up in this case, you can only perform `SELECT` operations. So, you can search but not add, modify, or delete data. 😏)
+
+## Regular LIKE search
+This is the method to perform a `LIKE` search using the standard functionality of PostgreSQL.
+
+(By the way, normally, indexes do not work for PostgreSQL's partial match, but if you create an index with PGroonga, it will work. It's like magic! 👀)
+
+### Search by title
+
+Open your browser and access the following:
+
+[`http://localhost:3000/memos?title=like.*データ*`](http://localhost:3000/memos?title=like.*データ*)
+
+```json
+[{"id":1,"title":"PostgreSQLはリレーショナル・データベース管理システムです。","content":"すごいでしょう"}]
+```
+
+### Search by content
+
+Open your browser and access the following:
+
+[`http://localhost:3000/memos?content=like.*ショウ*`](http://localhost:3000/memos?content=like.*ショウ*)
+
+```json
+[{"id":2,"title":"Groongaは日本語対応の高速な全文検索エンジンです。","content":"スワイショウ"}]
+```
+☝️ With the standard LIKE search functionality, searching for 'ショウ' in katakana will not match hiragana.
+
+## Searching with PGroonga
+
+Now, since the operators `&@~` required for searching with PGroonga are not directly available in PostgREST, we will create a stored function to enable their usage.
+
+```sh
+psql api
+```
+
+Execute the following SQL statement:
+
+```sql
+CREATE FUNCTION find_title(keywords text) RETURNS SETOF memos AS $$
+BEGIN
+  RETURN QUERY SELECT * FROM memos WHERE title &@~ keywords;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE FUNCTION find_content(keywords text) RETURNS SETOF memos AS $$
+BEGIN
+  RETURN QUERY SELECT * FROM memos WHERE content &@~ keywords;
+END;
+$$ LANGUAGE plpgsql;
+```
+
+## Searching with PGroonga in PostgREST
+When calling stored functions in PostgREST, the URL format is `/rpc/function_name`.
+
+Open your browser and access the following:
+
+[`http://localhost:3000/rpc/find_title?keywords=コマンド`](http://localhost:3000/rpc/find_title?keywords=コマンド)
+
+The following results will be returned.
+```json
+[{"id":4,"title":"groongaコマンドがあります。","content":"今日はコンバンワこのくにわ"}]
+```
+
+By the way, using a browser to hit the URL is easier than using curl because dealing with encoding can be cumbersome.
+
+```sh
+curl "http://localhost:3000/rpc/find_title?keywords=%E3%82%B3%E3%83%9E%E3%83%B3%E3%83%89"
```

Comment:
How about using `--get` and `--data-urlencode`?

```suggestion
curl --get --data-urlencode keywords=コマンド http://localhost:3000/rpc/find_title
```

Reply:
@askdkc ping

Reply:
Oh, I missed this comment 😅